boa/src/builtins/array/mod.rs

```diff
@@ -487,6 +488,45 @@ impl Array {
         Ok(a.into())
     }
 
+    ///'Array.prototype.at(index)'
+    ///
+    /// When at method takes desired integer as index and returns the value at given
+    /// index. Negative integer counts backwards. If index is invalid, the at method
+    /// returns undefined.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-array.prototype.at
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at
+    pub(crate) fn at(this: &JsValue, args: &[JsValue], context: &mut Context) -> JsResult<JsValue> {
+        //1. let O be ? ToObject(this value)
+        let obj = this.to_object(context)?;
+        //2. let len be ? LengthOfArrayLike(O)
+        let len = obj.length_of_array_like(context)? as i64;
+        //3. let relativeIndex be ? ToIntegerOrInfinity(index)
+        let relative_index = args.get(0).unwrap().to_integer_or_infinity(context)?;
```

Comment:
```suggestion
        let relative_index = args.get_or_undefined(0).to_integer_or_infinity(context)?;
```

This won't panic if `at` is called with no arguments, like so: `[1, 2, 3, 4].at()`

Reply:
I noticed this literally like 15-20 minutes ago! Completely unaware that get_or_undefined existed! Definitely making this update.